front/lib/api/vaults.ts

```diff
@@ -15,7 +18,22 @@ export const deleteVault = async (
   }
 
   const dataSourceViews = await DataSourceViewResource.listByVault(auth, vault);
-  // TODO(VAULTS_INFRA check if datasource is used in any agent configuration)
+
```

Comment:
So to be able to delete a datasource we sequentially compute the usage for each view? 🥵 
Not sure there's any other option as we want to display the name of the assistants using them but seems quite heavy to me..

Reply:
We don't have an index on vaultId so we either do it via each datasourceview or we take them all (accross all vaults) and then filter (on the js side or the sql side).
To me, it's more likely that a single vault has a limited amount of views.
Also, it's the delete action, it's okay if it takes some time imho, I updated the client side to show a loading btw.